Create a syntax highlighting editor for esc script

lavaduderDev opened the pull request:
Allows for esc scripts to be edited inside Godot instead of an external editor.

Currently it's very basic, it just highlights commands, and saves/loads files. 

No live error checking here, but esc script is pretty hard to mess up.

lavaduderDev commented:
Oh nuts, I did not mean to include my other pull request commits, just this one. WHOOPS!

fleskesvor commented:
No problem. What happened is that you merged master to update your branch, but since master is the working branch for your other PR, it has all those changes in it too.

What you usually want to do is to rebase your commits on master of this repository, but since you currently have a PR on your master branch, you'd have to create a new branch for that.

These changes shouldn't be difficult to merge though, so don't worry about that now. We just need to clean up this branch, so make sure you have it checked out:

    git checkout escedit

Next, we remove all those extra commits by moving HEAD back to point at your second commit 

    git reset 9a48a8d8756a6dd9ea08686e73401f79bd46c5a0

Then all you have to do is push your changes again, but now you need to use `--force`, since we have to overwrite the remote branch

    git push --force origin escedit

If you need any help, feel free to ask here or on the `#escoria` channel on [Freenode IRC](https://webchat.freenode.net/).

fleskesvor commented:
I didn't realise this PR was for the experimental esc editor, so those are the instructions for removing everything but the force quit option and editor hints. I'm not 100% sure what the best strategy here is. Normally, I'd make a new branch, reset this branch and cherry-pick commits back in, but it's probably not the best way, and might be a bit difficult to explain.

lavaduderDev commented:
Oh that doesn't sound good. I got to break this down. Okay so I have two branches (master, and escedit) that have been edited, and an backup branch.  
That have the following commits.

1. Made Configurable force quit option. 3-12-18
2. Made it easier to find .esc files. 3-12-18
3. Created a text editor for escoria 3/16/18 
4. added demo music for BG music 'backup branch, mixed in'
5. Added NPC to demo 'backup branch, mixed in'
6. updated graphics for player in demo 'backup branch, mixed in'

Not to mention all the remote merging with the upstream.

Part of the confusion is my inexperience (I don't know how to control git.), and the other is what you want. (I don't know what a clean Pull Request is for this repository.) 
First off tell me what you want for the pull requests on each commit. Second walk me through those steps.

Tell me what time you want me to get on godot's official freenode IRC (or at least a scheadual of your free time.)

Do you want me to just scrap the master branch entirely (With the exception of upstream merging)
And make branches that contain commits like so
esc_editor
 Created a text editor for escoria 3/16/18 
easyfiles
 Made it easier to find .esc files. 3-12-18
forcequit
 Made Configurable force quit option. 3-12-18
demoupdate
 added demo music for BG music 'backup branch, mixed in'
 Added NPC to demo 'backup branch, mixed in'
 updated graphics for player in demo 'backup branch, mixed in'
Then PR each of those branches?

lavaduderDev commented:
All commits have been reset. This should be good to go.

StraToN commented:
I was looking into PluginScript for this so it would enable the current Script editor to manage ESC scripts (with its own syntax as well) but I don't know if it is possible anyway with it. 

Great job! I'll try your branch tonight.

StraToN commented:
Ok I just tested the plugin. I have difficulties to open an existing esc script, I can't change the res://test.esc in the textbox.

lavaduderDev commented:
WHOOPS! I forgot to change the label to textedit. 

I just found out about `EditorFileDialog`. So that issue might be resolved entirely. Not sure how long it might be though, There's no Godot documentation on how to use it, It seems to be the same as Regular `FileDialog`.

StraToN commented:
> Okay so, the textedit had mouse ignore instead of mouse stop

Indeed. It worked after changing this setting.
It works indeed, even though it may be a bit rough for now. I'd like to check what we could do with ScriptPlugin before validating this.

StraToN commented:
We'll probably be able to use Godot's GDScript editor to edit any text file https://github.com/godotengine/godot/pull/19225 . If we do, this PR may need changes to just add syntax highlighting for ESC.
